bigtable/client/filters.h

```diff
@@ -449,6 +495,30 @@ class Filter {
   /// An empty filter, discards all data.
   Filter() : filter_() {}
 
+  /// Append @p head and @p tail to @p chain recursively.
```

Comment:
I'm skeptical this is better than just putting a for loop in the filter functions.

Reply:
Unfortunately there is no for-loop for parameter packs, and folding expressions for parameter packs do not show up until C++17.  On the other hand, these are compile-time recursions, so maybe not that horrible?

Reply:
I could easily be wrong about how this would work, but I was thinking you could use an initializer list and then iterate over the result?

`for (const auto& stage : {stages...}) { // add to Chain or Interleave }`

Reply:
Initializer lists only have const iterators, so we would need to copy each one of the filter protos out.  The for-loop is more readable, less calling overhead, the recursion is (in most cases) zero copy.

Using the for loop, thanks!